src/main/java/programminglife/gui/controller/GuiController.java

```diff
@@ -209,16 +233,22 @@ private void initLeftControlpanelDraw() {
         btnDraw.setOnAction(event -> {
             System.out.printf("%s Drawing graph...\n", Thread.currentThread());
 
-            int maxDepth = Integer.MAX_VALUE;
             int centerNode = 0;
+            int maxDepth = Integer.MAX_VALUE;
 
             try {
-                maxDepth = Integer.parseInt(txtMaxDrawDepth.getText());
                 centerNode = Integer.parseInt(txtCenterNode.getText());
+                maxDepth = Integer.parseInt(txtMaxDrawDepth.getText());
             } catch (NumberFormatException e) {
-                Alert alert = new Alert(Alert.AlertType.WARNING, "Input is not a number", ButtonType.OK);
+                Alert alert = new Alert(Alert.AlertType.WARNING, "Make sure you have entered a number as input.");
+                alert.show();
+            }
+
+            if (centerNode > this.graphController.getGraph().size()) {
```

Comment:
This is incorrect, as we cannot assume that the ids are sequential (you should create a method `contains` in `Graph`)